src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -58,6 +61,61 @@ trait DataUseRestrictionSupport extends LazyLogging {
 
   }
 
+  /**
+    * Looks at a DuosDataUse object (likely previously retrieved from ORSP) and translates it into a
+    * set of structured Data Use Restriction attributes.
+    *
+    * @param duosDataUse
+    * @return
+    */
+  def generateStructuredUseRestrictionAttribute(duosDataUse: DuosDataUse, ontologyDAO: OntologyDAO): Map[AttributeName, Attribute] = {
+    // these are straightforward mappings
+    val gru = duosDataUse.generalUse.map ( AttributeName.withLibraryNS("GRU") -> AttributeBoolean(_) )
+    val hmb = duosDataUse.hmbResearch.map ( AttributeName.withLibraryNS("HMB") -> AttributeBoolean(_) )
+    val rspd = duosDataUse.vulnerablePopulations.map ( AttributeName.withLibraryNS("RS-PD") -> AttributeBoolean(_) )
+    val ncu = duosDataUse.commercialUse.map ( AttributeName.withLibraryNS("NCU") -> AttributeBoolean(_) )
+    val nmds = duosDataUse.methodsResearch.map ( AttributeName.withLibraryNS("NMDS") -> AttributeBoolean(_) )
+
+    // FC supports both NCU and NPU; DUOS only supports one (NCU)
+    // val npu = duosDataUse.commercialUse.map ( AttributeName.withLibraryNS("NPU") -> AttributeBoolean(_) )
+
+    // DUOS represents NAGR and NCTRL as strings ("yes"|"no")
+    val nagr = duosDataUse.aggregateResearch match {
+      case Some(s) if "yes".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NAGR") -> AttributeBoolean(true))
+      case Some(s) if "no".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NAGR") -> AttributeBoolean(false))
+      case _ => None
+    }
+    val nctrl = duosDataUse.controlSetOption match {
+      case Some(s) if "yes".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NCTRL") -> AttributeBoolean(true))
+      case Some(s) if "no".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NCTRL") -> AttributeBoolean(false))
+      case _ => None
+    }
+
+    // DUOS has no concept of "IRB review required"
+    // val irb = None
+
+    val ds = duosDataUse.diseaseRestrictions match {
+      case Some(Seq()) => Map.empty[AttributeName,Attribute]
+      case Some(nodeList) =>
+        Map(
+          AttributeName.withLibraryNS("DS_URL") -> AttributeValueList(nodeList.map( AttributeString )),
+          AttributeName.withLibraryNS("DS") -> AttributeValueList(nodeList.map{ nodeid =>
+            ontologyDAO.search(nodeid) match {
+              case termResource :: Nil => AttributeString(termResource.label)
```

Comment:
Not specific to this PR, but I'm confused about why `search` returns a `List`.

Looking at the ES DAO, it can only be an empty list or a list with exactly one item.

Reply:
Good call - and the reason is just legacy. In the past, the implementation *could* return a list, and callers typically just used the head of it, sometimes threw an error if it returned >1. We changed the underlying implementation, but never changed the method signatures. That would be a good cleanup (outside of this PR, I propose)